pwiz_tools/Skyline/Model/Irt/IrtDbManager.cs

```diff
@@ -66,6 +66,10 @@ protected override bool StateChanged(SrmDocument document, SrmDocument previous)
             {
                 return true;
             }
+            if (!ReferenceEquals(document.Id, previous.Id))

```

Comment:
I am again not sure we want to force every implementation of BackgroundLoader to check for the document ID changing. This should be a very fast check. So probably not necessary to protect in the state change logic of the implementation. Better to just check it first in the code that calls StateChanged()